src/components/forms/common/AutocompleteDifficulties.tsx

```diff
@@ -1,38 +1,70 @@
-import { Box } from '@mui/material';
+import type { FC } from 'react';
+import { Box, Autocomplete, TextField } from '@mui/material';
 import LensIcon from '@mui/icons-material/Lens';
-import { AutocompleteElement } from 'react-hook-form-mui';
+import { Controller, type Control } from 'react-hook-form';
 
-import type { Difficulty } from '../../../utils/Difficulty';
-import { Difficulties } from '../../../utils/Difficulty';
+import { Difficulties, type Difficulty } from '../../../utils/Difficulty';
+
+type Props = {
+	control: Control<any>;
+	label: string;
+	name: string;
+};
+
+const getColor = (value: number) =>
+	[...Difficulties].find(difficulty => value === difficulty.value)?.color;
```

Comment:
Difficulties jsou seřazené, takže teoreticky by mohlo stačit zavolat `Difficulties[value - 1].color`, je to míň čitelné, ale asi efektivnější.